opm/simulators/flow/DamarisWriter.hpp

```diff
@@ -259,15 +269,81 @@ class DamarisWriter : public EclGenericWriter<GetPropType<TypeTag, Properties::G
                 this->damarisUpdate_ = false; 
             }
 
-            if (this->damarisOutputModule_->getPRESSURE_ptr() != nullptr) 
+            /*if (this->damarisOutputModule_->getPRESSURE_ptr() != nullptr) 
             {
                 dam_err_ = DamarisOutput::setPosition("PRESSURE", rank_,
                                                       this->elements_rank_offsets_[rank_]);
                 dam_err_ = DamarisOutput::write("PRESSURE", rank_,
                                                 this->damarisOutputModule_->getPRESSURE_ptr());
 
                 dam_err_ =  DamarisOutput::endIteration(rank_);
+            }*/
+            
+            
+            // Call damaris_set_position() for all available variables
+            // There is an assumption that all variables are the same size, with the same offset.
+            // see initDamarisTemplateXmlFile.cpp for the Damaris XML descriptions.
+            for ( auto damVar : localCellData ) {
+                // std::map<std::string, data::CellData>
+                const std::string name = damVar.first ;
+                dam_err_ = DamarisOutput::setPosition(name.c_str(), rank_,
+                                                      this->elements_rank_offsets_[rank_]);
+            }
+
+            // Call damaris_write() for all available variables
+            for ( auto damVar : localCellData ) 
+            {
+               // std::map<std::string, data::CellData>
+              const std::string& name = damVar.first ;
+              
+              std::unordered_set<std::string>::const_iterator is_in_set = wanted_vars_set_.find ( name );
+              
+              if ((is_in_set != wanted_vars_set_.end() ) || (wanted_vars_set_.size() == 0)) {
+                  data::CellData  dataCol = damVar.second ;
+                  OpmLog::debug(fmt::format("Name of Damaris Variable       : ( rank:{})  name: {}  ",  rank_, name));
+                  
+                  // It does not seem I can test for what type of data is present (double or int)
+                  // in the std::variant within the data::CellData, so I will use a try catch block. 
+                  // Although, only MPI_RANK and GLOBAL_CELL_INDEX are set as integer types (in the 
+                  // XML file) so it is a moot point. 
+                  // We could use damaris_get_type() to check what the type is specified as
+                  // within the Damaris XML file
```

Comment:
I see your point, usually std::holds_alternative could be used, but as the variant itself is not exposed that is not possible here. Maybe we should add an observer method to the data::CellData class? Alternatively, should we treat MPI_RANK and GLOBAL_CELL_INDEX specially to avoid this try-catch?

Reply:
Will there never be <int> values? If so, then using a variant seems redundant?

Reply:
I do not understand the comment.

I suggest the following changes. First on line 297, add:
```
const bool data_is_integer = (name == "MPI_RANK" || name == "GLOBAL_CELL_INDEX");
```
Then instead of the try, use the `data_is_integer` variable to select which code block to run.

Reply:
do you mean this comment: 
"Will there never be values? If so, then using a variant seems redundant?" or another one?
I think it may read better as:
 "Will there never be integer values? If so, then using a variant seems redundant?"